pytrickle/manager.py

```diff
@@ -0,0 +1,194 @@
+"""
+Base stream manager for trickle streaming applications.
+
+This module provides reusable stream management functionality that can be
+extended for specific use cases while maintaining core streaming patterns.
+"""
+
+import asyncio
+import logging
+from abc import ABC, abstractmethod
+from typing import Dict, Any, Optional, Protocol, Callable
+
+from .health import StreamHealthManager
+
+logger = logging.getLogger(__name__)
+
+class StreamHandler(Protocol):
+    """Protocol defining the interface that stream handlers must implement."""
+    
+    @property
+    def running(self) -> bool:
+        """Whether the stream handler is currently running."""
+        ...
+    
+    async def start(self) -> bool:
+        """Start the stream handler. Returns True if successful."""
+        ...
+    
+    async def stop(self, *, called_by_manager: bool = False) -> bool:
+        """Stop the stream handler. Returns True if successful."""
+        ...
+
+class BaseStreamManager(ABC):
+    """Base stream manager that provides core stream management functionality."""
+    
+    def __init__(self, health_manager: Optional[StreamHealthManager] = None):
+        self.handlers: Dict[str, StreamHandler] = {}
+        self.lock = asyncio.Lock()
+        self.health_manager = health_manager
+    
+    @abstractmethod
+    async def create_stream_handler(self, request_id: str, **kwargs) -> Optional[StreamHandler]:
```

Comment:
I am not seeing why we need this. I was thinking a stream would look the same from a trickle protocol perspective for all streams.

Reply:
I've had success with moving a ton of generic error handling logic for stream processing in https://github.com/livepeer/pytrickle/pull/9. I'm going to try building an app from these changes. @ad-astra-video lmk if you think we should go include this